core/Db/Schema/Mysql.php

```diff
@@ -781,6 +786,43 @@ public function getSupportedReadIsolationTransactionLevel(): string
         return 'READ UNCOMMITTED';
     }
 
+    public function hasReachedEOL(): bool
+    {
+        $currentVersion = $this->getVersion();
+
+        // Aurora is managed by AWS and is updated automatically if EOL, therefor we can ignore that here
+        if (str_contains(strtolower($currentVersion), 'aurora')) {
+            return false;
+        }
+
+        // End of security update for certain MySQL versions as of https://en.wikipedia.org/wiki/MySQL#Release_history
+
+        // Support for 8.0 LTS ends in April 2026
+        if (
+            version_compare($currentVersion, '8.0', '>=') &&
+            version_compare($currentVersion, '8.1', '<') &&
+            Date::today()->isEarlier(Date::factory('2026-05-01'))
+        ) {
+            return false;
+        }
+
+        // Support for 8.4 LTS ends in April 2032
+        if (
+            version_compare($currentVersion, '8.4', '>=') &&
+            version_compare($currentVersion, '8.5', '<') &&
+            Date::today()->isEarlier(Date::factory('2032-05-01'))
+        ) {
+            return false;
+        }
+
+        // Support for all other versions prior 9.3 already ended
```

Comment:
Same as above comment, just for better clarity
```suggestion
        // Support for all other versions prior to 9.3 (not covered by conditions above) already ended
```